cmd/telemetrygen/pkg/logs/config.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logs
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen/internal/common"
+)
+
+// Config describes the test scenario.
+type Config struct {
+	common.Config
+	NumLogs        int
+	Body           string
+	SeverityText   string
+	SeverityNumber int32
+	TraceID        string
+	SpanID         string
+}
```

Comment:
Would it be possible to make all/most of these fields private? (also applies to other signals)

Reply:
We could, at least unless someone comes with another use case for them to be exported. 

I tended to think about exporting it all such that the config values for the telemetrygen that are exported by cmd flags are also available as API variables. This way there's parity of expected functionality if you're using the API or the cmd call.

Reply:
Hm, I think keeping it private helps us design the API more carefully. I agree that functionality should be available in both ways, but _how_ to do it may be different than this.

Reply:
Totally fair, that would be a larger conversation regarding how this is designed. Anything we make private won't be able to be printed out by the logger when running the cmd. Something like option functions would work but would mildly complicate the pflag setup. 

That said, I can totally make most of the configs private if you'd prefer that for now.

Reply:
Since e2etest is an isolated go module here, making these private would break the ability to access them for e2e tests 😞